### ISSUE TITLE
Auto complete and company complete don't send output from rdm to emacs

### ISSUE THREAD

**Issue by deuscovrigus** [Andersbakken/rtags#612]
Similar to issue #609 , I have a slightly related issue which might be a tad more complicated to debug. 
I have a cmake project with some submodules. Completion and all rtag features work great in the submodules, however the completion feature fails  in the main project. Jump to definition, error underlining still work great.

When I attempt to call company-complete or auto-complete rdm acknowledges that completions have been found, i.e.
Found completions (26) in cache /home/prj/janus/src/components/pvr/internal/face3d/exe/render_face.cxx:350:19
AC behavior:
No visual popup occurs when attempting to complete and
evaling (rtags-ac-candidates) yields nil in the main project sources. The same eval in the submodule sources yields a list of completions, which are then passed to the visual popup :
(#("buffer" 0 6 (rtags-ac-type "CXXMethod" rtags-ac-full "cl_mem & buffer()")) ...
I have (setq ac-sources '(ac-source-rtags)) setup for autocomplete.
If I remove the ac-source-rtags and replace with the default junk ac-words-in-same buffers etc, the visual pop-up and auto-complete works, so that's why I think rtags doesn't pass the completions from rdm to ac.

Company behavior:
an attempt on completion in the main project source files yields an error trace this time:
Company: An error occurred in auto-begin
Company: Back-end company-rtags async timeout with args (candidates )
In the submodule source file the behavior works as expected.

I am at the latest commit of rtags 
I realize this is very project specific, but it used to work. Any ideas?

**Comment by deuscovrigus** [comment]
Ok I think I found the problem and maybe the solution be of some use. I edebugged rtags-ac and found where the completions never went from rdm to rtags-ac.
It turns out that this segment of rtags-ac-candidates() was returning nil:

(and pos (string= (car rtags-last-completions)
                               (rtags-current-location pos))

It was actually comparing the string
"/home/prj/janus/src/components/pvr/internal/face3d/exe/render_parameterized_face.cxx:422:44:"

with
 "/proj/janus/src/components/pvr/internal/face3d/exe/render_parameterized_face.cxx:422:44:"

obviously, this returned nil since the strings are equal. On my system though, these sources are the exact same since i have the following symlink defined:
lrwxrwxrwx 1 root root 10 Jun 22  2015 /proj -> /home/prj/

The true mystery is why rtags mixed in the main project sources with  /proj/janus and some with /home/prj/janus while in the submodules where completion was working , every source had /home/prj/janus/... so the string comparison test passed.

Obviously, I looked in the compile_commands.json and everything is with the prefix /home/prj/janus
I have no idea how rtags might have picked up on the symlink /proj that points to /home/prj , since it's definitely not in the compile commands.

Maybe it's Ubuntu being weird? Or is the rtags api that queries paths of the system and it went haywire by returning the symlink path rather than the actual path? 
I will obviously remove the symlink but it's still interesting to see what caused it.

**Comment by deuscovrigus** [comment]
Followup, i think this might be a user error, since ido-mode of emacs opened the file:
/proj/janus/src/components/pvr/internal/face3d/exe/render_parameterized_face (the symlink'ed path)
and that's why rtag-ac returned this string instead of the native path string pointed  by the symlink,i.e.
/home//prj/janus/src/components/pvr/internal/face3d/exe/"
As the saying goes: user error, get new user
Closing.

**Comment by Andersbakken** [comment]
Sorry about the delay in getting back to you. Good job debugging btw. I
actually think rtags should insulate the user from this behavior. Our
approach in general is to always resolve symlinks. I think we should do
that in elisp as well. I'll try to fix it.

Anders

**Comment by Andersbakken** [comment]
I added something that I hope will take care of this. Can you check?

**Comment by deuscovrigus** [comment]
Checked and confirmed! 
Experiment:
Made symbolic link:
lrwxrwxrwx  1 octi octi        16 Mar  3 15:27 rtags_fake -> /home/octi/rtags
Indexed the rtags source tree such that the database points to files in ~/rtags
Then I opened in emacs the source file with the path prefix being rtags_fake (which in effect points to rtags)
~/rtags_fake/src/ClangIndexer.cpp

Completion and features work great!

Thanks yet again Anders!
